src/scales/scale.time.js

```diff
@@ -539,30 +569,35 @@ module.exports = Scale.extend({
 
 	determineDataLimits: function() {
 		var me = this;
-		var adapter = me._adapter;
 		var options = me.options;
-		var tickOpts = options.ticks;
+		var adapter = me._adapter;
 		var unit = options.time.unit || 'day';
-		var min = Number.POSITIVE_INFINITY;
-		var max = Number.NEGATIVE_INFINITY;
-		var minmax = me._getMinMax(false);
-		var i, ilen, labels;
-
-		min = Math.min(min, minmax.min);
-		max = Math.max(max, minmax.max);
-
-		labels = getLabelTimestamps(me);
-		for (i = 0, ilen = labels.length; i < ilen; ++i) {
-			min = Math.min(min, labels[i]);
-			max = Math.max(max, labels[i]);
+		var {min, max, minDefined, maxDefined} = me._getUserBounds();
+
+		function _apply(bounds) {
```

Comment:
Can we call this `_applyBounds` just to make it a little clearer?

Reply:
done